### PR TITLE
Remove hardcoded clickhouse-client invocations from tests

### PR DESCRIPTION
**Pull request by mkmkme**
This PR is a followup to #57804. I checked all edited tests before pushing them.

Cc @Algunenano as a requestor of the changes :)

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit db43fb1004abd9b1c467d4a0480f71cad32a6614 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57861/db43fb1004abd9b1c467d4a0480f71cad32a6614/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7208761586">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57861/db43fb1004abd9b1c467d4a0480f71cad32a6614/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/queries/shell_config.sh [hunk 1]
```diff
@@ -27,7 +27,10 @@ export CLICKHOUSE_TEST_UNIQUE_NAME="${CLICKHOUSE_TEST_NAME}_${CLICKHOUSE_DATABAS
 export CLICKHOUSE_BINARY=${CLICKHOUSE_BINARY:="$(command -v clickhouse)"}
 # client
 [ -x "$CLICKHOUSE_BINARY-client" ] && CLICKHOUSE_CLIENT_BINARY=${CLICKHOUSE_CLIENT_BINARY:=$CLICKHOUSE_BINARY-client}
-[ -x "$CLICKHOUSE_BINARY" ] && CLICKHOUSE_CLIENT_BINARY=${CLICKHOUSE_CLIENT_BINARY:=$CLICKHOUSE_BINARY client}
+if [ -x "$CLICKHOUSE_BINARY" ]; then
```
Comment (Algunenano) [inline tests/queries/shell_config.sh:30]: This is wrong. `benchmark` should be treated separately (in its own block) and in a similar way as client/local/server...

**Review by Algunenano — CHANGES_REQUESTED** [review]

File: tests/queries/shell_config.sh (hunk 1 above)
Comment (mkmkme) [reply tests/queries/shell_config.sh]: Good point, thanks! Fixed

Comment (Algunenano) [reply tests/queries/shell_config.sh]: Nice, thank you!

File: tests/queries/shell_config.sh [hunk 2]
```diff
@@ -39,10 +39,13 @@ export CLICKHOUSE_LOCAL=${CLICKHOUSE_LOCAL:="${CLICKHOUSE_BINARY}-local"}
 [ -x "${CLICKHOUSE_BINARY}-server" ] && CLICKHOUSE_SERVER_BINARY=${CLICKHOUSE_SERVER_BINARY:="${CLICKHOUSE_BINARY}-server"}
 [ -x "${CLICKHOUSE_BINARY}" ] && CLICKHOUSE_SERVER_BINARY=${CLICKHOUSE_SERVER_BINARY:="${CLICKHOUSE_BINARY} server"}
 export CLICKHOUSE_SERVER_BINARY=${CLICKHOUSE_SERVER_BINARY:="${CLICKHOUSE_BINARY}-server"}
+# benchmark
+[ -x "${CLICKHOUSE_BINARY}-benchmark" ] && CLICKHOUSE_BENCHMARK_BINARY=${CLICKHOUSE_BENCHMARK_BINARY:="${CLICKHOUSE_BINARY}-benchmark"}
```
Comment (Algunenano) [inline tests/queries/shell_config.sh:43]:
```suggestion
[ -x "${CLICKHOUSE_BINARY}-benchmark" ] && CLICKHOUSE_BENCHMARK_BINARY=${CLICKHOUSE_BENCHMARK_BINARY:="${CLICKHOUSE_BINARY}-benchmark ${CLICKHOUSE_BENCHMARK_OPT0:-}"}
```

File: tests/queries/shell_config.sh [hunk 3]
```diff
@@ -39,10 +39,13 @@ export CLICKHOUSE_LOCAL=${CLICKHOUSE_LOCAL:="${CLICKHOUSE_BINARY}-local"}
...
+[ -x "${CLICKHOUSE_BINARY}" ] && CLICKHOUSE_BENCHMARK_BINARY=${CLICKHOUSE_BENCHMARK_BINARY:="${CLICKHOUSE_BINARY} benchmark"}
```
Comment (Algunenano) [inline tests/queries/shell_config.sh:44]:
```suggestion
[ -x "${CLICKHOUSE_BINARY}" ] && CLICKHOUSE_BENCHMARK_BINARY=${CLICKHOUSE_BENCHMARK_BINARY:="${CLICKHOUSE_BINARY} benchmark ${CLICKHOUSE_BENCHMARK_OPT0:-}"}
```

**Review by Algunenano — CHANGES_REQUESTED** [review]

File: tests/queries/shell_config.sh (hunk 2 above)
Comment (mkmkme) [reply tests/queries/shell_config.sh:43]: Hmmmm, maybe in that case ditch `CLICKHOUSE_BENCHMARK_BINARY` completely and assign `CLICKHOUSE_BENCHMARK` already here?

IMHO having `CLICKHOUSE_BENCHMARK_OPT0` for `_BINARY` is a bit inconsistent at least with `CLICKHOUSE_CLIENT`, where `CLICKHOUSE_CLIENT_BINARY` is just a path to the binary itself, and `CLICKHOUSE_CLIENT` contains all the needed options.

File: tests/queries/shell_config.sh (hunk 3 above)
Comment (mkmkme) [reply tests/queries/shell_config.sh:44]: same question here

File: tests/queries/shell_config.sh (hunk 2 above)
Comment (Algunenano) [reply tests/queries/shell_config.sh:43]: Yep, makes sense. We just need to make sure `CLICKHOUSE_BENCHMARK` contains the options

Comment (mkmkme) [reply tests/queries/shell_config.sh:43]: Ah yeah, I overlooked that it added the options *only*  if `CLICKHOUSE_BENCHMARK_BINARY` hadn't been set earlier. Thanks, I'll fix it!

**Comment by mkmkme** [comment]
@Algunenano in the latest commit I made the benchmark handling the same as client one. Does it look okay?

**Review by Algunenano — APPROVED** [review]
LGTM. It's likely that some places could use `$CLICKHOUSE_CLIENT` instead of `$CLICKHOUSE_CLIENT_BINARY` but at least now it's not hardcoded

**Comment by mkmkme** [comment]
Can it be merged then? This diff changes too many files, I'm a bit afraid of possible conflicts :)

**Comment by Algunenano** [comment]
Yes, but test failures need to be reviewed first, and that takes time.

**Comment by Algunenano** [comment]
Failures:
* `Integration tests (asan) [4/4] ` -> test_storage_azure_blob_storage -> https://github.com/ClickHouse/ClickHouse/issues/57889
* `Integration tests (release) [4/4]` -> test_storage_azure_blob_storage -> https://github.com/ClickHouse/ClickHouse/issues/57889
* `Stateless tests (release, s3 storage) [2/2]`. Timeouts in `SYSTEM FLUSH LOGS` (https://github.com/ClickHouse/ClickHouse/issues/57984)
* `Stress test (tsan)` -> https://github.com/ClickHouse/ClickHouse/issues/57894
* `Stress test (ubsan)` -> https://github.com/ClickHouse/ClickHouse/issues/57892 (Reverted. Coincidentally, it's a feature you added and had to be removed)

**Comment by mkmkme** [comment]
> Coincidentally, it's a feature you added and had to be removed
Yup, I noticed this 😅 Trying to fix the data race rn

**Pull request merged**